exchanges/bitget/bitget.go

```diff
@@ -0,0 +1,5033 @@
+	errPlanTypeEmpty                  = errors.New("planType cannot be empty")
+	errPlanOrderIDEmpty               = errors.New("planOrderID cannot be empty")
+	errHoldSideEmpty                  = errors.New("holdSide cannot be empty")
+	errExecutePriceEmpty              = fmt.Errorf("%w: ExecutePrice below minimum", limits.ErrPriceBelowMin)
+	errIDListEmpty                    = errors.New("idList cannot be empty")
+	errLoanTypeEmpty                  = errors.New("loanType cannot be empty")
+	errProductIDEmpty                 = errors.New("productID cannot be empty")
+	errPeriodTypeEmpty                = errors.New("periodType cannot be empty")
+	errLoanCoinEmpty                  = fmt.Errorf("%w: LoanCoin is required", currency.ErrCurrencyCodeEmpty)
+	errCollateralCoinEmpty            = fmt.Errorf("%w: CollateralCoin is required", currency.ErrCurrencyCodeEmpty)
+	errTermEmpty                      = errors.New("term cannot be empty")
+	errCollateralAmountEmpty          = fmt.Errorf("%w: CollateralAmount below minimum", limits.ErrAmountBelowMin)
+	errCollateralLoanMutex            = errors.New("exactly one of collateralAmount and loanAmount must be set")
+	errReviseTypeEmpty                = errors.New("reviseType cannot be empty")
+	errUnknownPairQuote               = errors.New("unknown pair quote; pair can't be split due to lack of delimiter and unclear base length")
+	errStrategyMutex                  = errors.New("only one of immediate or cancel, fill or kill, and post only can be set to true")
+	errReturnEmpty                    = errors.New("returned data unexpectedly empty")
+	errAuthenticatedWebsocketDisabled = errors.New("authenticatedWebsocketAPISupport not enabled")
+	errAssetModeEmpty                 = errors.New("assetMode cannot be empty")
+	errTakeProfitTriggerPriceEmpty    = fmt.Errorf("%w: TakeProfitTriggerPrice below minimum", limits.ErrPriceBelowMin)
+	errStopLossTriggerPriceEmpty      = fmt.Errorf("%w: StopLossTriggerPrice below minimum", limits.ErrPriceBelowMin)
+	errOrderIDMutex                   = errors.New("exactly one of orderID and clientOrderID must be set")
+	errProductTypeAndPairEmpty        = errors.New("productType and pair cannot both be empty")
+
+	prodTypes = []string{"USDT-FUTURES", "COIN-FUTURES", "USDC-FUTURES"}
+	planTypes = []string{"normal_plan", "track_plan", "profit_loss"}
+)
+
+// QueryAnnouncements returns announcements from the exchange, filtered by type and time
+func (e *Exchange) QueryAnnouncements(ctx context.Context, annType string, startTime, endTime time.Time, pagination int64, limit uint8) ([]AnnouncementResp, error) {
+	var params Params
+	params.Values = make(url.Values)
+	err := params.prepareDateString(startTime, endTime, true, true)
+	if err != nil {
+		return nil, err
+	}
+	params.Values.Set("annType", annType)
+	params.Values.Set("language", "en_US")
+	params.Values.Set("idLessThan", strconv.FormatInt(pagination, 10))
+	if limit != 0 {
```

Comment:
check if the `limit` is `> 0`, not `!= 0` the use `strconv.FormatInt()` instead.